src/flow_solver/flow_solver_cases/non_periodic_cube_flow.cpp

```diff
@@ -28,11 +57,72 @@ std::shared_ptr<Triangulation> NonPeriodicCubeFlow<dim,nstate>::generate_grid()
 template <int dim, int nstate>
 void NonPeriodicCubeFlow<dim,nstate>::display_additional_flow_case_specific_parameters() const
 {
-    // Do nothing for now.
+    this->pcout << "- - Courant-Friedrichs-Lewy number: " << this->all_param.flow_solver_param.courant_friedrichs_lewy_number << std::endl;
+}
+
+template <int dim, int nstate>
+double NonPeriodicCubeFlow<dim,nstate>::get_adaptive_time_step(std::shared_ptr<DGBase<dim,double>> dg) const
+{
+    // compute time step based on advection speed (i.e. maximum local wave speed)
+    const unsigned int number_of_degrees_of_freedom_per_state = dg->dof_handler.n_dofs()/nstate;
+    const double approximate_grid_spacing = (this->all_param.flow_solver_param.grid_right_bound-this->all_param.flow_solver_param.grid_left_bound)/pow(number_of_degrees_of_freedom_per_state,(1.0/dim));
+    const double cfl_number = this->all_param.flow_solver_param.courant_friedrichs_lewy_number;
+    const double time_step = cfl_number * approximate_grid_spacing / this->maximum_local_wave_speed;
+    
+    return time_step;
+}
+
+template <int dim, int nstate>
+double NonPeriodicCubeFlow<dim,nstate>::get_adaptive_time_step_initial(std::shared_ptr<DGBase<dim,double>> dg)
+{
+    // initialize the maximum local wave speed
+    update_maximum_local_wave_speed(*dg);
+    // compute time step based on advection speed (i.e. maximum local wave speed)
+    const double time_step = get_adaptive_time_step(dg);
+    return time_step;
+}
+
+template<int dim, int nstate>
+void NonPeriodicCubeFlow<dim, nstate>::update_maximum_local_wave_speed(DGBase<dim, double> &dg)
+{    
+    // Initialize the maximum local wave speed to zero
+    this->maximum_local_wave_speed = 0.0;
```

Comment:
if these are intended to be the same as those in `PeriodicTurbulence`; can you find a way to call those?

Reply:
this way, we won't ever have to fix code in two places

Reply:
@jbrillon , do you think we could move some of these common functions (e.g., adaptive time stepping) to `flow_solver_case_base`?

Reply:
@cpethrick only `update_maximum_local_wave_speed` could be moved since the others rely on the approximate_grid_spacing calculation which is currently specific to uniform grids. Another solution would be to derive NonPeriodicCubeFlow from PeriodicTurbulence and override and function defition needed

Reply:
@ssaishruthi98 Maybe we could have an intermediate class called `PeriodicCubeFlow_UniformGrid` where we move the definitions for `get_adaptive_time_step()`, `get_adaptive_time_step_initial()`, and `update_maximum_local_wave_speed()`. Then `PeriodicTurbulence` and `NonPeriodicCubeFlow` would be derived from `PeriodicCubeFlow_UniformGrid`.

Reply:
Created an intermediate class called `CubeFlow_UniformGrid` and moved the definitions for the three functions to that class. `PeriodicTurbulence` and `NonPeriodicCubeFlow` now use the functions in `CubeFlow_UniformGrid`! 😄